src/main/java/com/techconative/inmemory/pagination/util/InmemoryFOPS.java

```diff
@@ -0,0 +1,253 @@
+package com.techconative.inmemory.pagination.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.techconative.inmemory.pagination.modal.OrderingCriteria;
+import com.techconative.inmemory.pagination.modal.PageResult;
+import com.techconative.inmemory.pagination.modal.PaginationCriteria;
+import exception.InvalidCriteriaException;
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Filter, search, sort and paginate your data. PaginationService is the class to be extended to
+ * utilise the library
+ *
+ * @see <a href="https://github.com/techconative/inmemory-pagination">Git repository</a> for usage.
+ */
+@Slf4j
+public class InmemoryFOPS {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * The central method which gets the data and apply filter, search, sort and paginates based on
+     * criteria
+     *
+     * @param rawData input data to apply criteria operations on
+     * @param criteria contains filter, search, sort and pagination constraints
+     * @return List of result data with constraints applied
+     * @since 1.0.0
+     */
+    public static PageResult processData(List rawData, PaginationCriteria criteria) {
+        List<Map<String, String>> data = convert(rawData);
+
+        if (criteria.getFilter() != null && !criteria.getFilter().isEmpty()) {
+            data = applyFiltering(criteria, data);
+        }
+
+        LinkedList<Map<String, String>> sortedList = applySorting(criteria, data);
+
+        LinkedList resultList = applyPagination(criteria, sortedList);
```

Comment:
I'm not looking into the logics inside as I assumed these logics are moved from one class to other.

Reply:
Yes. Beyond a few minor edits most are just move to the new class.